parsec/_parsec_pyi/protocol.pyi

```diff
@@ -0,0 +1,64 @@
+from typing import Optional
+from parsec._parsec_pyi.ids import BlockID, RealmID
+
+class AuthenticatedAnyCmdReq:
+    def __repr__(self) -> str: ...
+    def dump(self) -> bytes: ...
+    def load(buf: bytes) -> BlockReadReq: ...
+
+class InvitedAnyCmdReq:
+    def __repr__(self) -> str: ...
+    def dump(self) -> bytes: ...
+    def load(buf: bytes) -> BlockReadReq: ...
+
+class BlockCreateReq:
+    def __init__(self, block_id: BlockID, realm_id: RealmID, block: bytes) -> None: ...
+    def __repr__(self) -> str: ...
+    def __eq__(self, other: BlockCreateReq) -> bool: ...
+    def __ne__(self, other: BlockCreateReq) -> bool: ...
+    @property
+    def block_id(self) -> BlockID: ...
+    @property
+    def realm_id(self) -> RealmID: ...
+    @property
+    def block(self) -> bytes: ...
+    def dump(self) -> bytes: ...
+
+class BlockCreateRep:
+    def __repr__(self) -> str: ...
+    def __eq__(self, other: BlockCreateRep) -> bool: ...
+    def __ne__(self, other: BlockCreateRep) -> bool: ...
+    def is_ok(self) -> bool: ...
+    def Ok() -> BlockCreateRep: ...
+    def AlreadyExists() -> BlockCreateRep: ...
```

Comment:
This is really cool 😎

Reply:
How are handled the unkown error ?

for instance if the server returns a `{"status: "unknown_command"}` ?

cf. the generic errors statuses that can raise the server no matter what is the req:

https://github.com/Scille/parsec-cloud/blob/f9639c0982f47c811711098d4532c4c87da59ae6/parsec/backend/asgi/ws.py#L176-L217

On top of that the server may be more recent than the client, in this case it might return a status that we don't understand but which is valid (adding error type is considered a minor API bump)

(I remember we say that we should have a special type representing an unknown error for this case)

Reply:
Aaaaand we should have a test that ensure unknown status are correctly handled by those new kind of schema classes (typically add it into `tests/api/test_protocol.py`

Reply:
UnknownError no longer exists